suite-common/wallet-core/src/transactions/transactionReducer.ts

```diff
@@ -0,0 +1,145 @@
+                const { account, txs } = payload;
+                const transactions = state.transactions[account.key] || [];
+                txs.forEach(tx => {
+                    const index = transactions.findIndex(t => t.txid === tx.txid);
+                    transactions.splice(index, 1);
+                });
+            })
+            .addCase(transactionActions.addTransaction, (state, { payload }) => {
+                const { transactions, account, page } = payload;
+                if (transactions.length < 1) return;
+                initializeAccount(state, account.key);
+                const accountTxs = state.transactions[account.key];
+
+                if (!accountTxs) return;
+
+                transactions.forEach((transaction, i) => {
+                    // first we need to make sure that transaction is not undefined, then check if transactionid matches
+                    const existingTx = findTransaction(transaction.txid, accountTxs);
+                    if (!existingTx) {
+                        // add a new transaction
+                        if (page) {
+                            // insert a tx object at correct index
+                            const txIndex = (page - 1) * settingsCommonConfig.TXS_PER_PAGE + i;
+                            accountTxs[txIndex] = transaction;
+                        } else {
+                            // no page arg, insert the tx at the beginning of the array
+                            accountTxs.unshift(transaction);
+                        }
+                    } else {
+                        // update the transaction if conditions are met
+                        const existingTxIndex = accountTxs.findIndex(
+                            t => t && t.txid === existingTx.txid,
+                        );
+
+                        if (
+                            (!existingTx.blockHeight && transaction.blockHeight) ||
+                            (!existingTx.blockTime && transaction.blockTime)
+                        ) {
+                            // pending tx got confirmed (blockHeight changed from undefined/0 to a number > 0)
+                            accountTxs[existingTxIndex] = { ...transaction };
+                        }
+                    }
+                });
+            })
+            .addCase(accountsActions.removeAccount, (state, { payload }) => {
+                payload.forEach(a => {
+                    delete state.transactions[a.key];
+                });
+            })
+            .addMatcher(
+                action => action.type === extra.actionTypes.storageLoad,
+                extra.reducers.storageLoadTransactions,
+            )
+            .addMatcher(
+                action => action.type === extra.actionTypes.fiatRateUpdate,
+                extra.reducers.txFiatRateUpdate,
+            );
+    },
+);
+
+export const selectIsTransactionsLoading = (state: TransactionRootState) =>
```

Comment:
It's good convetion to keep "is" and "Loading" together. => `selectIsLoadingTransactions`